src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,25 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a Contact parent class in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
+ */
+public class Contact {
```

Comment:
Glad to see that header comments are added =).
Could you think of a way for validation to be done in `Contact` class instead of `Address`, `Phone` and `Email`?